core/src/apps/common/cbor.py

```diff
@@ -217,6 +233,29 @@ def __eq__(self, other: object) -> bool:
         )
 
 
+class Set:
+    """Spec taken from: https://github.com/input-output-hk/cbor-sets-spec/blob/master/CBOR_SETS.md"""
+    def __init__(self, array: List[Value]) -> None:
+        self.raise_if_duplicates(array)
+        self.array = array
+
+    def raise_if_duplicates(self, array):
+        for i in range(0, len(array)):
+            for j in range(0, len(array)):
+                if i == j:
+                    continue
+                if array[i] == array[j]:
+                    raise ValueError
```

Comment:
1. Of course, thanks.
2. I wasn't sure whether this was possible and I haven't looked into it, since the other ValueErrors also don't contain messages. It makes sense of course.